dev/validate_bom__test.py

```diff
@@ -420,6 +430,35 @@ def __init__(self, deployer):
         'echo-worker': 8089
     }
 
+    # Map of services with provided hosts and credentials.
+    outbound_service_configs = {}
+    if self.options.test_gate_service_host:
+      gate_service_config = {
+        'host': self.options.test_gate_service_host,
+      }
+      bearer_auth_token = self.__gate_service_bearer_auth_token_or_none()
```

Comment:
```
gate_credentials_path = self.options.test_gate_iap_credentials
if gate_credentials_path:
  gate_service_config['service_account_email'] = get_service_account_email(gate_credentials_path)
  client_id = self.options.test_gate_iap_client_id
  auth_token = self.__bearer_auth_token_or_none('gate', gate_credentials_path, client_id)
  if auth_token:
    gate_service_config['bearer_auth_token'] = auth_token
```
And you can wrap 435..443 into a add_gate_service_config(configs) function that
```
if not self.options.test_gate_service_host:
   return
436...444 but can remove "gate_" from local variables.
```

Then replace 435...444 with self.add_gate_service_config(outbond_service_configs)